modules/ExperimentBrowser/src/client/ExperimentBrowser.coffee

```diff
@@ -623,8 +627,55 @@ class ExperimentBrowserController extends Backbone.View
 				code = @experimentController.model.get('lsLabels').getLabelByTypeAndKind('id', 'study id')[0].get('labelText')
 			else
 				code = @experimentController.model.get('codeName')
-
-			window.open("/openExptInQueryTool?experiment=#{code}",'_blank')
+			# Add Generating Link Loading Mask 
+			@$('.bv_generatingLink').show()
+			# Call to Route to Get URL 
+			$.ajax
+				type: 'GET'
+				url: "/getLinkExptQueryTool?experiment=#{code}"
+				success: (response) => 
+					window.open(response,'_blank')
+				error: (err) =>
+					console.log err
+				datatype: 'json'
+			# Take Away Generating Progress Mask 
+			@$('.bv_generatingLink').hide()
+
+	handleGetLinkQueryToolClicked: => 
+			# Preparatory Logic to Get Code to Pass to Generate Link Route 
+			if @experimentController.model.get('lsLabels') not instanceof LabelList
+				@experimentController.model.set 'lsLabels',  new LabelList @experimentController.model.get('lsLabels')
+			subclass = @experimentController.model.get('subclass')
+			if window.conf.entity?.saveInitialsCorpName? and window.conf.entity.saveInitialsCorpName is true
+				if @experimentController.model.get('lsLabels').getLabelByTypeAndKind("corpName", subclass + ' corpName').length > 0
+					code = @experimentController.model.get('lsLabels').getLabelByTypeAndKind('corpName', subclass + ' corpName')[0].get('labelText')
+				else if @experimentController.model.get('lsKind') is "study" and @experimentController.model.get('lsLabels').getLabelByTypeAndKind('id', 'study id').length > 0
+					code = @experimentController.model.get('lsLabels').getLabelByTypeAndKind('id', 'study id')[0].get('labelText')
+				else
+					code = @experimentController.model.get("codeName")
+			else if @experimentController.model.get('lsKind') is "study" and @experimentController.model.get('lsLabels').getLabelByTypeAndKind('id', 'study id').length > 0
+				code = @experimentController.model.get('lsLabels').getLabelByTypeAndKind('id', 'study id')[0].get('labelText')
+			else
+				code = @experimentController.model.get('codeName')
+			# Add Generating Link Loading Mask 
+			@$('.bv_generatingLink').show()
+			# Call to Route to Get URL 
+			$.ajax
+				type: 'GET'
+				url: "/getLinkExptQueryTool?experiment=#{code}"
+				success: (response) => 
+					@$('.bv_getLinkResults').show()
+					@$('.bv_exptLink').val(response)
+				error: (err) =>
+					console.log err
+				datatype: 'json'
+			# Take Away Generating Progress Mask 
+			@$('.bv_generatingLink').hide()
+
+	handleCopyLinkClicked: => 
+		@$('.bv_exptLink').select()
+		document.execCommand("copy")
```

Comment:
I don't think we want to use this as it's using a deprecated browser feature:
https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand

You should be able to replace this with:

```
handleCopyLinkClicked: =>
    link = @$('.bv_exptLink').value
    if link? # Defined and not null
            alert("Link copied to clipboard!")
    else
            alert("Unable to copy link to clipboard")
```

I don't really like us using `alerts` in this way but I think the alternative might be to complicated to implement right now.  Prefer to just show another element and then have it fade away after a few seconds.  Alerts are very in your face I think and generally not used unless we really need to block the user from something (i.e. from moving off the page because of unsaved data).

Reply:
Makes sense. Thanks for the background info. It really helps!